libbeat/processors/uuid/generator/elasticsearch/generator.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elasticsearch
+
+import (
+	"encoding/base64"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var (
+	sequenceNumber uint32
+	lastTimestamp  *time.Time
+	mac            []byte
+	mu             sync.Mutex
+)
+
+func init() {
+	m, err := getSecureMungedMACAddress()
+	if err != nil {
+		panic(err)
+	}
+	mac = m
+	sequenceNumber = rand.Uint32()
+}
```

Comment:
Although I'm no fan of global vars, I see why we need it here (processors have no support for shared state). Yet I'd prefer if we replace init in favor of lazy initialization (using `sync.Once`). Reason is getSecureMungedMACAddress uses the `crypt/rand` package which might try to read random symbols provided by the OS. During system startup the amount of entropy might be limited, making the randome initialization potentially fail.

Plus `rand.Uint32` might always give you the same value. We set the seed in some init function, but if this init function is called before the one initializing the rand package we will run this call with a seed of 0. As we depend on init execution order as has been decided by the compiler, the seed is not much predictable now and in the future.